### PR TITLE
CLDC-763

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Refactor creation of page titles into a method

### CODE REVIEW

**Comment by paulrobertlloyd** [comment]
@kosiakkatrina I think we can move the helper to the application layout, then pages only need to give the title text.

**Comment by kosiakkatrina** [comment]
> @kosiakkatrina I think we can move the helper to the application layout, then pages only need to give the title text.

@paulrobertlloyd is this what you meant https://github.com/communitiesuk/mhclg-data-collection-beta/pull/136/commits/53b76d38c2760a2736201aaa671e6cbd5a470e87 ?

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html lang="en" class="govuk-template">
   <head>
-    <title><%= yield(:title).present? ? yield(:title) : t('service_name') %></title>
+    <title><%= yield(:title).present? ? browser_title(yield(:title)) : t('service_name') %></title>
```
Comment (paulrobertlloyd) [inline app/views/layouts/application.html.erb]: This is close, but if there’s no `:title`, the browser title will be `[Service name]` not `[Service name] - GOV.UK`.

I wonder if we can use the helper like this here:

```suggestion
    <title><%= browser_title(yield(:title)) %></title>
```

And then in the helper, if a title is present, output:

`[title] - [Service name] - GOV.UK`

and if no title is given, output:

`[Service name] - GOV.UK`

**Comment by paulrobertlloyd** [comment]
❤️ it!

**Pull request merged**